docs/release_notes/v0.14.0.rst

```diff
@@ -0,0 +1,54 @@
+New in 0.14.0 (2021-10-04)
+--------------------------
+
```

Comment:
let's add a paragraph here in plain english mentioning the user guide, the new lda model, and the setup anndata change with a code example (before -> after) in the case of one particular model